kernel/src/table_changes/scan_file.rs

```
@@ -127,26 +129,35 @@ impl<T> RowVisitor for CdfScanFileVisitor<'_, T> {
+                        .unwrap_or(0);
+                    let deletion_vector = visit_deletion_vector_at(row_index, &getters[2..=6])?;
+                    let partition_values = getters[7]
                         .get_opt(row_index, "scanFile.add.fileConstantValues.partitionValues")?;
-                    (scan_type, path, deletion_vector, partition_values)
-                } else if let Some(path) = getters[7].get_opt(row_index, "scanFile.remove.path")? {
+                    (scan_type, path, size, deletion_vector, partition_values)
+                } else if let Some(path) = getters[8].get_opt(row_index, "scanFile.remove.path")? {
                     let scan_type = CdfScanFileType::Remove;
-                    let deletion_vector = visit_deletion_vector_at(row_index, &getters[8..=12])?;
-                    let partition_values = getters[13].get_opt(
+                    let size = getters[9]
+                        .get_opt(row_index, "scanFile.remove.size")?
+                        .unwrap_or(0);
+                    let deletion_vector = visit_deletion_vector_at(row_index, &getters[10..=14])?;
+                    let partition_values = getters[15].get_opt(
                         row_index,
                         "scanFile.remove.fileConstantValues.partitionValues",
                     )?;
-                    (scan_type, path, deletion_vector, partition_values)
-                } else if let Some(path) = getters[14].get_opt(row_index, "scanFile.cdc.path")? {
+                    (scan_type, path, size, deletion_vector, partition_values)
+                } else if let Some(path) = getters[16].get_opt(row_index, "scanFile.cdc.path")? {
                     let scan_type = CdfScanFileType::Cdc;
-                    let partition_values = getters[15]
+                    let size = getters[17]
+                        .get_opt(row_index, "scanFile.cdc.size")?
+                        .unwrap_or(0);
```

If it's still possible to pass 0 as the size, doesn't our parquet reader code still need to know how to deal with it, and we _shouldn't_ delete the HEAD request dance that avoids suffix requests in Azure?

---

But actually, the [Delta spec](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-cdc-file) says cdc size column is required, so we should just error out if it's missing:
> size |	Long |	The size of this file in bytes

(sidecar `sizeInBytes` is also mandatory)